lib/ensure/link.js

```diff
@@ -1,39 +1,39 @@
 'use strict'
 
-const u = require('universalify').fromCallback
+const u = require('universalify').fromPromise
 const path = require('path')
-const fs = require('graceful-fs')
+const fs = require('../fs')
 const mkdir = require('../mkdirs')
-const pathExists = require('../path-exists').pathExists
+const { pathExists } = require('../path-exists')
 const { areIdentical } = require('../util/stat')
 
-function createLink (srcpath, dstpath, callback) {
-  function makeLink (srcpath, dstpath) {
-    fs.link(srcpath, dstpath, err => {
-      if (err) return callback(err)
-      callback(null)
-    })
+async function createLink (srcpath, dstpath) {
+  let dstStat
+  try {
+    dstStat = await fs.lstat(dstpath)
+  } catch {
+    // ignore error
   }
 
-  fs.lstat(dstpath, (_, dstStat) => {
-    fs.lstat(srcpath, (err, srcStat) => {
-      if (err) {
-        err.message = err.message.replace('lstat', 'ensureLink')
-        return callback(err)
-      }
-      if (dstStat && areIdentical(srcStat, dstStat)) return callback(null)
+  let srcStat
+  try {
+    srcStat = await fs.lstat(srcpath)
+  } catch (err) {
+    err.message = err.message.replace('lstat', 'ensureLink')
+    throw err
+  }
+
+  if (srcStat && dstStat && areIdentical(srcStat, dstStat)) return
```

Comment:
Again, we're throwing in the catch block, so no need to check for the existence of the variable:
```suggestion
  if (dstStat && areIdentical(srcStat, dstStat)) return
```

Reply:
Fixed in [`fbfc9c7` (#1023)](https://github.com/jprichardson/node-fs-extra/pull/1023/commits/fbfc9c7d86925105cf905df9e11b51132be386d4).